src/lib.rs

```diff
@@ -256,7 +273,7 @@ impl SweepGradientSource {
         phi = if_then_else(xabs < yabs, 1.0/4.0 - phi, phi);
         phi = if_then_else(px < 0.0   , 1.0/2.0 - phi, phi);
         phi = if_then_else(py < 0.0   , 1.0 - phi     , phi);
-        phi = if_then_else(phi != phi , 0.              , phi);  // Check for NaN.
```

Comment:
phi != phi doesn't compile with no_std?

Reply:
not related I don't think, but rust-analyzer was giving an error: 
![image](https://github.com/jrmuizel/sw-composite/assets/79148527/2bc97287-4da0-476f-b6ff-fe8b7467b2ff)